relay/shfu_storage/sqlite_dialect.go

```diff
@@ -0,0 +1,112 @@
+package shfu_storage
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+)
+
+const sqliteTimeFormat = "2006-01-02 15:04:05"
+
+// SQLiteDialect implements DBDialect for SQLite databases
+type SQLiteDialect struct{}
+
+// NewSQLiteDialect creates a new SQLite dialect
+func NewSQLiteDialect() *SQLiteDialect {
+	return &SQLiteDialect{}
+}
+
+// GetCreateTableSQL returns SQLite-specific table creation statements
+func (d *SQLiteDialect) GetCreateTableSQL() []string {
+	return []string{
+		`CREATE TABLE IF NOT EXISTS shfu_records (
+			chain_id TEXT NOT NULL,
+			counterparty_chain_id TEXT NOT NULL,
+			from_height_revision_number INTEGER NOT NULL,
+			from_height_revision_height INTEGER NOT NULL,
+			to_height_revision_number INTEGER NOT NULL,
+			to_height_revision_height INTEGER NOT NULL,
+			updated_at TEXT NOT NULL, -- SQLite has no native DATETIME type, uses TEXT for dates
+			update_client_results BLOB,
+			latest_finalized_header BLOB, -- Serialized core.Header bytes
+			PRIMARY KEY (chain_id, counterparty_chain_id, from_height_revision_number, from_height_revision_height, to_height_revision_number, to_height_revision_height)
+		)`,
+		`CREATE INDEX IF NOT EXISTS idx_shfu_records_chain 
+		 ON shfu_records(chain_id)`,
+		`CREATE INDEX IF NOT EXISTS idx_shfu_records_height 
+		 ON shfu_records(to_height_revision_height)`,
+		`CREATE INDEX IF NOT EXISTS idx_shfu_records_updated_at 
+		 ON shfu_records(updated_at)`,
+	}
+}
+
+// ConvertTimeToDB converts Go time to SQLite DATETIME format
+func (d *SQLiteDialect) ConvertTimeToDB(t time.Time) interface{} {
+	// Use SQLite's preferred ISO 8601 format for DATETIME columns
+	// This enables proper date/time comparisons and functions
+	return t.UTC().Format("2006-01-02 15:04:05")
```

Comment:
`sqliteTimeFormat` can be used here:

```suggestion
	return t.UTC().Format(sqliteTimeFormat)
```

Reply:
fixed